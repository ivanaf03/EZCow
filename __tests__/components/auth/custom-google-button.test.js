import React from "react";

import { render, fireEvent } from "@testing-library/react-native";

import CustomGoogleButton from "../../../components/basic/custom-button";

describe("CustomGoogleButton", () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it("should render correctly", () => {
    const tree = render(<CustomGoogleButton text="Google" onPress={() => {}} />);
    expect(tree).toMatchSnapshot();
  });

  it("should call onPress when pressed", () => {
    const onPress = jest.fn();
    const tree = render(<CustomGoogleButton text="Google" onPress={onPress} />);
    fireEvent.press(tree.getByText("Google"));
    expect(onPress).toHaveBeenCalledTimes(1);
  });
});
