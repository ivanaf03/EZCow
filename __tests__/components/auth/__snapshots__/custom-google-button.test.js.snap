// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomGoogleButton should render correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onBlur={[Function]}
  onClick={[Function]}
  onFocus={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    [
      {
        "display": "flex",
      },
      {
        "flexDirection": "row",
      },
      {
        "justifyContent": "space-between",
      },
      {
        "alignItems": "center",
      },
      {
        "paddingBottom": 16,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 16,
      },
      {
        "backgroundColor": "#272727",
      },
      {
        "borderBottomWidth": 2,
        "borderLeftWidth": 2,
        "borderRightWidth": 2,
        "borderTopWidth": 2,
      },
      {
        "borderBottomColor": "#6bc0eb",
        "borderLeftColor": "#6bc0eb",
        "borderRightColor": "#6bc0eb",
        "borderTopColor": "#6bc0eb",
      },
      {
        "borderBottomLeftRadius": 24,
        "borderBottomRightRadius": 24,
        "borderTopLeftRadius": 24,
        "borderTopRightRadius": 24,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontFamily": "Nunito-SemiBold",
        },
        {
          "color": "#ffffff",
        },
        {
          "fontSize": 20,
          "lineHeight": 28,
        },
        {
          "textAlign": "center",
        },
      ]
    }
  >
    Google
  </Text>
</View>
`;
